cmd/rig/cmd/capsule/root/status.go

```diff
@@ -0,0 +1,277 @@
+package root
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"connectrpc.com/connect"
+	"github.com/fatih/color"
+	"github.com/rigdev/rig-go-api/api/v1/capsule"
+	"github.com/rigdev/rig-go-api/api/v1/capsule/rollout"
+	"github.com/rigdev/rig-go-api/operator/api/v1/pipeline"
+	"github.com/rigdev/rig/cmd/common"
+	capsule_cmd "github.com/rigdev/rig/cmd/rig/cmd/capsule"
+	"github.com/rigdev/rig/cmd/rig/cmd/flags"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+var boldWhite = color.New(color.FgWhite, color.Bold)
+var red = color.New(color.FgRed)
+
+func (c *Cmd) status(ctx context.Context, cmd *cobra.Command, _ []string) error {
+	statusResp, err := c.Rig.Capsule().GetStatus(ctx, &connect.Request[capsule.GetStatusRequest]{
+		Msg: &capsule.GetStatusRequest{
+			CapsuleId:     capsule_cmd.CapsuleID,
+			ProjectId:     flags.GetProject(c.Scope),
+			EnvironmentId: flags.GetEnvironment(c.Scope),
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	status := statusResp.Msg.GetStatus()
+	if flags.Flags.OutputType != common.OutputTypePretty {
+		return common.FormatPrint(status, flags.Flags.OutputType)
+	}
+
+	printStatusSummary(status, capsule_cmd.CapsuleID)
+
+	return nil
+}
+func printStatusSummary(s *capsule.Status, capsuleID string) {
+	str := buildCapsuleInfo(s, capsuleID)
+	str = str + buildRolloutStatus(s.GetRollout())
+	str = str + buildContainerConfig(s.GetContainerConfig())
+	str = str + buildInstanceStatus(s.GetInstances())
+	str = str + buildConfigFileStatus(s.GetConfigFiles())
+	str = str + buildInterfaceStatus(s.GetInterfaces())
+	str = str + buildCronjobStatus(s.GetCronJobs())
+	fmt.Print(str)
+
+}
+
+func buildCapsuleInfo(s *capsule.Status, capsuleID string) string {
+	str := boldWhite.Sprintf("Capsule %s\n", capsuleID)
+	return str + boldWhite.Sprintf("Namespace: %s\n", s.GetNamespace())
+}
+
+func buildRolloutStatus(r *capsule.RolloutStatus) string {
```

Comment:
In general with these methods, passing a `string.Builder` to the functions and instead of just string-concatenation prevents the constant string allocation